From: Andrea Bolognani <eof@kiyuko.org>
Date: Mon, 30 Jan 2012 17:50:15 +0100
Subject: Use secure temp file for printing

The contents of the current note have to be saved on a temp file
for printing. Instead of using a well-defined file in the user's
home directory, create a new temp file each time, and remove it
after printing.

Forwarded: Marv Boyes <greyspace@tuxfamily.org>
---
 rhinote.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/rhinote.py b/rhinote.py
index b56f938..75a90d5 100644
--- a/rhinote.py
+++ b/rhinote.py
@@ -25,6 +25,7 @@ import tkinter
 import tkinter.filedialog as tkFileDialog
 import tkinter.messagebox as tkMessageBox
 import os
+import tempfile
 
 # the root window:
 def Rhinote():
@@ -81,12 +82,14 @@ class TextWidget(tkinter.Text):
         Rhinote()
 
     def printfile(self, whatever = None):
-        f = open(self.printfilename, 'w')
+        # Use a secure temp file for printing
+        f = tempfile.NamedTemporaryFile(prefix = 'Rhinoteprintfile-', delete = False)
         f.write(self.get('1.0', 'end'))
-        f.close
+        f.close()
         # 'enscript' formats the text; lpr sends it to the default printer;
         # enscript's -B option suppresses page headers.
-        os.system('enscript -B --word-wrap $HOME/.Rhinoteprintfile > lpr &')
+        os.system('enscript -B --word-wrap %s > lpr &' % f.name)
+        os.unlink(f.name)
 
     def help(self, whatever = None):
         tkMessageBox.showinfo('Rhinote Help', message = '''
@@ -128,7 +131,6 @@ http://rhinote.tuxfamily.org
         self.bind('<Control-H>', self.help)
         self.master = master
         self.filename = ''
-        self.printfilename = os.environ['HOME']+'/.Rhinoteprintfile'
         self._filetypes = [
         ('Text/ASCII', '*.txt'),
         ('Rhinote files', '*.rhi'),
